### ISSUE TITLE
False positive in excessive_precision for floats that are already in shortest form

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#7721]
```rust
fn main() {
    let _ = 0.123_456_789_012_345_67;
}
```

```console
warning: float has excessive precision
 --> src/main.rs:2:13
  |
2 |     let _ = 0.123_456_789_012_345_67;
  |             ^^^^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.123_456_789_012_345_66`
  |
  = note: `#[warn(clippy::excessive_precision)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#excessive_precision
```

Clippy's suggestion is the following diff:

```diff
-     let _ = 0.123_456_789_012_345_67;
+     let _ = 0.123_456_789_012_345_66;
```

It seems wrong to call this excessive precision when both representations have identical precision. Probably there is a reason that the one the human wrote is more meaningful in context and should be left as is.

<br>

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.57.0-nightly (addb4da68 2021-09-25)
binary: rustc
commit-hash: addb4da686a97da46159f0123cb6cdc2ce3d7fdb
commit-date: 2021-09-25
host: x86_64-unknown-linux-gnu
release: 1.57.0-nightly
LLVM version: 13.0.0
```
